### PR TITLE
Feat/add pre commit hooks

### PR DESCRIPTION
**Pull request by Adrok24**
I added pre-commit file and some minor fixes.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          - flake8-comprehensions
+          - flake8-simplify
+          - flake8-typing-imports
+
+  - repo: https://github.com/nbQA-dev/nbQA
+    rev: 1.6.0
+    hooks:
+      - id: nbqa-black
+      - id: nbqa-isort
+      - id: nbqa-flake8
```
Comment (zigaLuksic) [inline .pre-commit-config.yaml:52]: this can probably be removed, as I dont think we'll be using any notebooks here

File: pyproject.toml [hunk 2]
```diff
@@ -38,3 +36,11 @@ min-public-methods = 0
...
 [tool.pylint.similarities]
 min-similarity-lines = 5
+
+[tool.pylint.classes]
+valid-metaclass-classmethod-first-arg = "mcs"
+
+[tool.nbqa.addopts]
+flake8 = [
+    "--extend-ignore=E402"
+]
```
Comment (zigaLuksic) [inline pyproject.toml:46]: also probably not needed

File: requirements-dev.txt [hunk 3]
```diff
@@ -1,7 +1,7 @@
-pb_tool
 black
```
Comment (zigaLuksic) [inline requirements-dev.txt:1]: you can add pre-commit and remove black/isort/flake8

File: pyproject.toml [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
 line-length = 120
 preview = true
+extend-exclude = "SentinelHub/resources.py|SentinelHub/external|SentinelHub/zip_build"
 
 [tool.isort]
 profile = "black"
 known_first_party = "sentinelhub"
 sections = ["FUTURE", "STDLIB", "THIRDPARTY", "LOCALFOLDER"]
 line_length = 120
+skip = ["SentinelHub/resources.py", "SentinelHub/external", "SentinelHub/zip_build"]
```
Comment (zigaLuksic) [inline pyproject.toml:11]: could you perhaps explain these two a bit? Not sure why they are added

Comment (zigaLuksic) [reply pyproject.toml:11]: what is the `resources.py` file? I somehow can't seem to find it, I only see the `resources.qrc`. Is this a dynamically generated file or am I searching in the wrong place?

Comment (Adrok24) [reply pyproject.toml]: Yes, the resources.qrc is compiled into a resources.py file with pyqt. When you run the deploy you create that file. The external is a folder containing the wheels files, and the zip contains the full plugin to be installed in the Qgis.

Comment (zigaLuksic) [reply pyproject.toml]: ah, so the exclusion is not done for the CI, but for the developers, which likely have a local `resources.py` file?

**Review by zigaLuksic — APPROVED** [review]

**Pull request merged**